gxexternalproviders/src/main/java/com/genexus/db/driver/ExternalProviderIBM.java

```diff
@@ -288,31 +304,31 @@ public void renameDirectory(String directoryName, String newDirectoryName) {
         newDirectoryName = StorageUtils.normalizeDirectoryName(newDirectoryName);
         createFolder(newDirectoryName);
         ListObjectsRequest listObjectsRequest = new ListObjectsRequest()
-                .withBucketName(bucket).withPrefix(directoryName);
+            .withBucketName(bucket).withPrefix(directoryName);
         for (S3ObjectSummary file : client.listObjects(listObjectsRequest).getObjectSummaries()) {
             String newKey = file.getKey().replace(directoryName, newDirectoryName);
-            rename(file.getKey(), newKey, false);
+            rename(file.getKey(), newKey, null);
         }
         deleteDirectory(directoryName);
     }
 
     public List<String> getFiles(String directoryName, String filter) {
-    	filter = (filter == null || filter.isEmpty())? null: filter.replace("*", "");
+        filter = (filter == null || filter.isEmpty())? null: filter.replace("*", "");
         List<String> files = new ArrayList<String>();
         directoryName = StorageUtils.normalizeDirectoryName(directoryName);
         ListObjectsRequest listObjectsRequest = new ListObjectsRequest()
-                .withBucketName(bucket).withPrefix(directoryName).withDelimiter(StorageUtils.DELIMITER);   
-        for (S3ObjectSummary file : client.listObjects(listObjectsRequest).getObjectSummaries()) {        	
-        	String key = file.getKey();        	
+            .withBucketName(bucket).withPrefix(directoryName).withDelimiter(StorageUtils.DELIMITER);
+        for (S3ObjectSummary file : client.listObjects(listObjectsRequest).getObjectSummaries()) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.